app/src/main/java/com/sopt/now/compose/feature/auth/LoginActivity.kt

```diff
@@ -77,48 +84,37 @@ class LoginActivity : ComponentActivity() {
         val intent = Intent(this@LoginActivity, SignUpActivity::class.java)
         resultLauncher.launch(intent)
     }
-
-    private fun handleLoginState(
-        idTextField: String,
-        pwdTextField: String
-    ) {
-        if (userData?.id.isNullOrEmpty() && userData?.password.isNullOrEmpty()) {
-            toast(getString(R.string.login_first_sign_up_null))
-            return
-        }
-        when {
-            userData?.id != idTextField -> {
-                toast(getString(R.string.login_not_id_equal))
-            }
-
-            userData?.password != pwdTextField -> {
-                toast(getString(R.string.login_not_pwed_equal))
-            }
-
-            else -> {
-                navigateToMainActivity(userData)
-                toast(getString(R.string.login_completed))
-            }
-        }
-    }
-
-    private fun navigateToMainActivity(userData: User?) {
-        Intent(this@LoginActivity, MainActivity::class.java).apply {
-            putExtra(KeyStorage.USER_INPUT, userData)
-        }.also {
-            startActivity(it)
-            finish()
-        }
-    }
 }
 
 @Composable
 fun LoginScreen(
     onLoginBtnClick: (String, String) -> Unit,
-    onSignUpBtnClick: () -> Unit
+    onSignUpBtnClick: () -> Unit,
+    viewModel: LoginViewModel
 ) {
     var id by remember { mutableStateOf("") }
     var password by remember { mutableStateOf("") }
+
+    val context = LocalContext.current
+    val loginState by viewModel.loginResponseState.collectAsState(initial = UiState.Loading)
+
+    LaunchedEffect(loginState) {
+        viewModel.loginResponseState.collect { uiState ->
+            when (uiState) {
+                is UiState.Success -> {
+                    Toast.makeText(context, uiState.data.toString(), Toast.LENGTH_SHORT).show()
+                    navigateToMainActivity(context, uiState.data)
+                }
+
+                is UiState.Failure -> {
+                    Toast.makeText(context, uiState.errorMessage, Toast.LENGTH_SHORT).show()
+                }
+
+                else -> Unit
+            }
+        }
+    }
```

Comment:
[P1]
collect는 최초 한 번만 수행해도 괜찮을 것 같습니다.

```suggestion

    LaunchedEffect(Unit) {
        viewModel.loginResponseState.collect { uiState ->
            when (uiState) {
                is UiState.Success -> {
                    Toast.makeText(context, uiState.data.toString(), Toast.LENGTH_SHORT).show()
                    navigateToMainActivity(context, uiState.data)
                }

                is UiState.Failure -> {
                    Toast.makeText(context, uiState.errorMessage, Toast.LENGTH_SHORT).show()
                }

                else -> Unit
            }
        }
    }
```

Reply:
감사합니다! 이렇게 해야 리컴포지션 안일어나더 라구요ㅠㅠ
계속 inspector돌려봤네요..

Reply:
[f73a049](https://github.com/NOW-SOPT-ANDROID/chanwoo-bae/pull/17/commits/f73a04900df19745f70149cdee8e57b6539e7b77) 감사합니다! 수정했습니당